plugins/modules/proxmox_kvm.py

```diff
@@ -1464,7 +1474,7 @@ def main():
             if vm['status'] == 'stopped':
                 module.exit_json(changed=False, vmid=vmid, msg="VM %s is not running" % vmid, **status)
 
-            if proxmox.stop_vm(vm, force=module.params['force']) and proxmox.start_vm(vm):
+            if proxmox.restart_vm(vm):
                 module.exit_json(changed=True, vmid=vmid, msg="VM %s is restarted" % vmid, **status)
         except Exception as e:
```

Comment:
Instead of using and except catch here I would rewrite the code to use try/catch inside of the `restart_vm` function for those API calls. It helps when you need to write unit tests.

Reply:
Great suggestion and have made the change. This suggestion also applies to the other states and will have to be considered in future pull requests.